File: CarritodeCompras/logica/Notificacion.cs
using System;

namespace CarritodeCompras.logica
{
    public class Notificacion
    {
        private string destinatario;
        private string mensaje;
        private DateTime fecha;

        public Notificacion(string destinatario, string mensaje)
        {
            this.destinatario = destinatario;
            this.mensaje = mensaje;
            this.fecha = DateTime.Now;
        }

        public string Destinatario { get { return destinatario; } }
        public string Mensaje { get { return mensaje; } }

        public void Enviar()
        {
            Console.WriteLine($"[Notificación para {destinatario} - {fecha:dd/MM/yyyy HH:mm}] {mensaje}");
        }
    }
}

File: CarritoBlueJ/CarritodeCompras/Notificacion.java

public class Notificacion
{
    private String destinatario;
    private String mensaje;
    private String fecha;

    public Notificacion(String destinatario, String mensaje, String fecha)
    {
        this.destinatario = destinatario;
        this.mensaje = mensaje;
        this.fecha = fecha;
    }

    public String getDestinatario()
    {
        return destinatario;
    }

    public String getMensaje()
    {
        return mensaje;
    }

    public void enviar()
    {
        System.out.println("[Notificación para " + destinatario + " - " + fecha + "] " + mensaje);
    }

    public String toString()
    {
        return "Para " + destinatario + ": " + mensaje;
    }
}